nomad/node_endpoint.go

```diff
@@ -1695,6 +1695,12 @@ func (n *Node) createNodeEvals(node *structs.Node, nodeIndex uint64) ([]string,
 		}
 		jobIDs[alloc.JobNamespacedID()] = struct{}{}
 
+		// If it's a sysbatch job, skip it. Sysbatch job evals should only
+		// ever be created by structs.EvalTriggerPeriodicJob
```

Comment:
Hm, I'm having a hard time tracking down the relationship between `sysbatch` and `EvalTriggerPeriodicJob`. sysbatch jobs aren't _necessarily_ periodic.

I think the following logic is appropriate: non-periodic sysbatch jobs are "one shot": whatever nodes are eligible when they're scheduled, are the nodes they run on. We don't guarantee that nodes that were `down` when the sysbatch job was submitted ever receive the job if they come back and re-register.

Reply:
Right! So the comment needs some re-phrasing I think, but the logic still holds. If it's a system batch job, and it's periodic, it'll get triggered by `periodic-job` and should not be triggered by anything else. If it's not periodic, `job-register` is the appropriate trigger and again, I believe the system scheduler should not be called on node update events.